### PR TITLE
LPS-138436 Filter not present collections to work with deleted collections

### PR DESCRIPTION
**Pull request by victorg1991**

### CODE REVIEW

**Comment by victorg1991** [comment]
ci:test:sf

**Comment by victorg1991** [comment]
ci:test:relevant

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CollectionFilterGeneralPanel.js [hunk 1]
```diff
@@ -241,10 +245,12 @@ function filterSupportedFilters({
...
-	const targetCollectionsSupportedFilters = targetCollections.map(
-		(targetCollection) =>
-			filterableCollections[targetCollection]?.supportedFilters || []
-	);
+	const targetCollectionsSupportedFilters = targetCollections
+		.map(
+			(targetCollection) =>
+				filterableCollections[targetCollection]?.supportedFilters
+		)
+		.filter((supportedFilters) => !!supportedFilters);
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CollectionFilterGeneralPanel.js:253]: If think we prefer `Boolean(supportedFilters)` (once :candle:Greg:candle: mentioned that to me). Although I do prefer `!!`, that's a triple operation that casts a value to boolean, and then flips the boolean twice. `Boolean(bla)` just casts the value and is technically faster.

Comment (victorg1991) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CollectionFilterGeneralPanel.js:253]: Changed! :)

**Comment by victorg1991** [comment]
ci:forward

**Comment by jkappler** [comment]
ci:forward

**Pull request closed without merging**